### PR TITLE
Fix adamw formula doc

### LINKED ISSUE #68482: Documentation for AdamW is wrong
## 🐛 Bug

The second line of the formula for `AdamW` here: https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html is wrong

## To Reproduce

Read the line `theta_t = theta_t  - gamma lambda theta_{t-1}`.

## Expected behavior

This should say `g_t = g_t - lambda theta_{t-1}`

It is clear that updating `theta_t` at this step doesn't make sense, since it isn't read again in the code.

Changing the update to `g_t` makes the algorithm match Adam https://pytorch.org/docs/stable/generated/torch.optim.Adam.html (with lambda != 0).

The paper referred to in the documentation: https://arxiv.org/pdf/1711.05101.pdf
Does it like this
`g_t = g_t + lambda theta_{t-1}`,
but the difference of +/- on lambda probably doesn't matter much.

cc @brianjo @mruberry @vincentqb @jbschlosser @albanD

### PR DESCRIPTION
**Pull request by albanD**
Fixes https://github.com/pytorch/pytorch/issues/68482

### CODE REVIEW

**Review by jbschlosser — COMMENTED** [review]
Hm I'm not sure I agree that this is the right change here - looks like the existing formula already matches the [paper](https://arxiv.org/pdf/1711.05101.pdf) notation in using `\theta_{t-1}`. What am I missing?

**Comment by albanD** [comment]
The difference is that the paper does not consider weight decay which we have here.
In this case, the weight decay updates based on t-1 first and then the step itself uses this new value.

**Review by jbschlosser — APPROVED** [review]
LGTM
